index.js

```diff
@@ -156,6 +168,34 @@ async function getActionId (code) {
   return row ? row.id : null
 }
 
+async function getAllDataForDownload (code) {
+  return query(`SELECT 
+    to_char(timestamp::date,'DD/MM/YYYY') AS date,  
+    timestamp::time(0) AS time,  
+    company.name AS company, 
+    pseudonym, 
+    action.code, 
+    action.description
+  FROM action_log
+  INNER JOIN company ON action_log.company_id=company.id
+  INNER JOIN action ON action_log.action_id = action.id;`)
+}
+
+async function getCompanyDataForDownload (companyName) {
+  return query(`SELECT 
+    to_char(timestamp::date,'DD/MM/YYYY') AS date,  
+    timestamp::time(0) AS time,  
+    company.name AS company, 
+    pseudonym, 
+    action.code, 
+    action.description
+  FROM action_log
+  INNER JOIN company ON action_log.company_id=company.id
+  INNER JOIN action ON action_log.action_id = action.id
+  WHERE company.name ILIKE '${companyName}';`

```

Comment:
⚠️ **SQL Injection alert!**
If somebody sent the string `'; DROP TABLE company; --` it would be destructive. That's why the `$1` syntax exists, it will make sure things are properly escaped. So this should be:
```
WHERE company.name ILIKE $1';`, [companyName])
```